### PR TITLE
feat(service-lifecycle): add type icon

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-1172

Screenshot 2024-07-02 at 15 47 34

Screenshot 2024-07-02 at 15 47 43

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Programs must be written for people to read, and only incidentally for machines to execute

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] test-remi2 - Feat/lifecycle job UI type frt 1172 - 2024-07-02T13:47:53Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/2773fa2b-2696-4d78-ac18-7633d72e5003/services/general)

 Another comment will be posted when deployments are finished

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [\[PR\] test-remi2 - Feat/lifecycle job UI type frt 1172 - 2024-07-02T13:47:53Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/2773fa2b-2696-4d78-ac18-7633d72e5003/logs)

File: libs/domains/services/feature/src/lib/general-setting/general-setting.tsx [hunk 1]
```diff
@@ -38,6 +40,21 @@ export function GeneralSetting({ label = 'Name' }: GeneralSettingProps) {
           />
         )}
       />
+      {type && (
```
Comment (ctjhoa) [inline libs/domains/services/feature/src/lib/general-setting/general-setting.tsx:43]: I think you can keep the original component and introduce in each `useForm` data where `GeneralSettings` is been used a new `template_type` property.
You can then set `defaultValues` of each forms depending on the service type, creating an alias on `schedule.lifecycle_type` if the service is a job.
We can then supports any service template.

File: libs/domains/services/feature/src/lib/service-badge/service-badge.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+export interface ServiceBadgeProps
+  extends Omit<ComponentPropsWithoutRef<'div'>, 'size'>,
+    VariantProps<typeof serviceBadgeVariants> {
+  icon: keyof typeof IconEnum
+  type?: keyof typeof JobLifecycleTypeEnum
+}
+
+// TODO: Prepare component to update service icon with custom icon
+export const ServiceBadge = forwardRef<ElementRef<'div'>, ServiceBadgeProps>(function ServiceBadge(
```
Comment (ctjhoa) [inline libs/domains/services/feature/src/lib/service-badge/service-badge.tsx:46]: I think we can improve several points on this component:
1. It's probably the right time to introduce a new UI component `ResourceAvatar` which can represent any resource visual marker. We can have typescript props the guarantee correct typing.
2. I don't like the term `Badge` as it's already something else in the design system.
3. A more composable approach is to have a component `Indicator`, which is only a wrapper to its children and add an indicator at its top right corner. Like the color dot we have but not limited to that case and supports of icons.

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/domains/services/feature/src/lib/general-setting/general-setting.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/domains/services/feature/src/lib/general-setting/general-setting.tsx]: https://github.com/Qovery/console/pull/1465/commits/03007b3e64976bef9151df12b8c4951951f95b24
https://github.com/Qovery/console/pull/1465/commits/a2a138c0cf82fd0c42655aeee0078f17c329344c

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**